### PR TITLE
lvm2: 2.02.177 -> 2.03.01

### PR DESCRIPTION
**Pull request by markuskowa**
###### Motivation for this change
Version bump. Switched the source to the git repository. Should now be fit for the auto updater.

CC @7c6f434c 

###### Things done
* switch to sources to git
* add `libaio` to `buildInputs`
* Fix grub (fallout from xfsprogs-4.19 update).

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by 7c6f434c** [comment]
@GrahamcOfBorg test installer.lvm

**Comment by 7c6f434c** [comment]
@GrahamcOfBorg biuld nixosTests.installer.lvm

**Comment by 7c6f434c** [comment]
And this is the first bump across what they called a set of large refactorings on the mailing list, so probably we shouldn't hurry with backporting?

**Comment by markuskowa** [comment]
What do you mean by backporting?
@GrahamcOfBorg build nixosTests.installer.lvm

**Comment by 7c6f434c** [comment]
I meant `release-18.09`

**Comment by markuskowa** [comment]
OK, if there is any need for backporting, there is also lvm2-2.02.183.

**Comment by 7c6f434c** [comment]
I have some doubts even about 2.02.183

**Comment by markuskowa** [comment]
IIRC there were also other problems with the installer tests, such as randomly timed out tests.

**Comment by 7c6f434c** [comment]
@pbogdan Well, there was also that `spice` failure breaking all the tests…

**Comment by markuskowa** [comment]
I just did a rebase. This should eliminate the issue with `spice` and leave only the xfsprogs/grub-os-prober problem on x86_64-linux.
@GrahamcOfBorg build nixosTests.installer.lvm

**Comment by markuskowa** [comment]
Now the test times out on x86_64-linux.

**Comment by 7c6f434c** [comment]
Locally it failed for GRUB reasons without this PR. #51790

**Comment by markuskowa** [comment]
With the XFS patch the lvm installer test succeeds again (locally, with ofborg it times out).

File: pkgs/tools/misc/grub/2.0x.nix [hunk 1]
```diff
@@ -82,7 +82,16 @@ stdenv.mkDerivation rec {
...
     '';
 
-  patches = [ ./fix-bash-completion.patch ];
+  patches = [
+    ./fix-bash-completion.patch
+    # This patch makes grub compatible with the XFS sparse inode
+    # feature introduced by xfsprogs-4.16.
+    # to be removed in grub-2.03
+    (fetchpatch {
+      url = http://git.savannah.gnu.org/cgit/grub.git/patch/?id=cda0a857dd7a27cd5d621747464bfe71e8727fff;
```
Comment (7c6f434c) [inline pkgs/tools/misc/grub/2.0x.nix]: Doesn't it work as https:// ?

Comment (markuskowa) [reply pkgs/tools/misc/grub/2.0x.nix]: Yes, works.

**Comment by hedning** [comment]
Test works locally here too :+1: any reason not to merge? at the very worse we'll break an already broke trunk :)

**Comment by markuskowa** [comment]
If there are no further comments it can be merged.

**Comment by hedning** [comment]
Hmm, seems like `2.03.00` removed  applib, the flag is no longer [recognized](https://hydra.nixos.org/build/85820869/nixlog/1):
> configure: WARNING: unrecognized options: --disable-static, --enable-applib

The lvm2 [commit](https://sourceware.org/git/?p=lvm2.git;a=commit;h=b67ef90438052d3e01f563cdd25e2268f4e02c20) removing it.

[osquery](https://hydra.nixos.org/build/85820869) breaks due to the removal:
> .../block_devices.cpp:24:10: fatal error: lvm2app.h: No such file or directory

`osquery` master [still depends on this](https://github.com/facebook/osquery/blob/master/osquery/tables/system/linux/block_devices.cpp#L24), so it's not enough to upgrade `osquery`.

cc latest `osquery` maintainer @Ma27

**Comment by markuskowa** [comment]
The lvm update also breaks `udisks1` and `udisks-glue`. The latest versions of those packages are from 2013 and 2014 and look rather outdated. Can they be removed (CC @pSub)?

**Comment by Ma27** [comment]
Even if it would be fixed on `oquery` master, it would take some time until I do the next `osquery` bump, unfortunately the diff between the patched cmake (I'm partially responsible for) and master's cmake is way too big now.

Until this is fixed upstream and I do the upgrade for `nixpkgs`, wouldn't it be possible to maintain two versions of the package (as this seems to be a breaking change)?

**Comment by 7c6f434c** [comment]
Reinstating 2.02.183 seems to be a natural plan…

**Comment by markuskowa** [comment]
So far it looks like  `osquery` is the only package not ready lvm2-2.03.x. Makes sense to add 2.02.183 as a legacy version. We can keep a legacy version until `osquery` is updated.

**Pull request merged**